design: motion + layout + ui/ux improvements

thomasyuill-livekit opened the pull request:
## Motion, Layout + UI/UX Improvements

This PR introduces several enhancements to improve the overall user experience and visual consistency across the application. This is not the final state, but a step in the right direction.

| iframe | popover |
|---|---|
|<img width="3200" height="1800" alt="localhost_3000__tab=iframe(16_9) (1)" src="https://github.com/user-attachments/assets/dfcab29c-562f-4182-8487-5c737ff668ae" /> | <img width="3200" height="1800" alt="localhost_3000__tab=iframe(16_9)" src="https://github.com/user-attachments/assets/e3f9ea2d-8ba6-4b5d-bab6-79ffa2bf140e" />|

- Migrated Header from React starter template
- Migrated Footer from React starter template
- Moved theme script application to the main app page
- Fixed color scheme with improved accent colors (--fgAccent: #1fd5f9, --bgAccent: #051518)
- Improved theme script injection with cleaner implementation
- Simplified session end and error messages
- Polished trigger button and added enter animation
- Merged "end call" into trigger button
- Improved responsive layout and z-indexing
- Simplified and polished the BarVisualizer state transitions
- Refined the transcript scroll behaviour

## Testing Notes

- Verify theme switching works correctly across all embed types
- Test responsive behavior on various screen sizes
- Ensure animations perform well
- Validate accessibility improvements with screen readers

bcherry commented:
this is very cool! we should make the popup version the default in the preview page...

thomasyuill-livekit commented:
@bcherry I polished it some more this morning
There's a few more visual improvements

thomasyuill-livekit commented on components/welcome.tsx:
```diff
@@ -1,88 +1,145 @@
+                  weight="regular"
+                  className="mt-0.5 inline shrink-0 rotate-[145deg] animate-bounce"
+                />
+              </div>
+            </div>
+            <EmbedPopupAgentClient appConfig={APP_CONFIG_DEFAULTS} />
+          </>
+        )}
```
I think there are several topics to discuss in this comment. I'll try to address them independently in brief, and we can have a deeper discussion about some of the more involved decisions.

## static/fixed positioning

I removed static, because the redesigned welcome screen no longer needed.
I do agree that users will want some flexibility here, but I think we can address that in a follow up PR.

I see 2 options
- render statically as a default, allow users to render in a positioned container
- provide app configuration that sets which corner and offset

##  EmbedPopupAgentClient vs embed-popup.js

I understand the difference, 
but for the purposes of demoing the experience on the welcome screen,
using `EmbedPopupAgentClient` was easier to implement than manipulating the DOM to remove the widget when switching tabs.

I agree we should have a conversation about the path forward, but I'm not sure I introduced anything that didn't already exist. So maybe this a fix forward thing, once we've aligned on the abstractions we want to expose. 

I think there more opportunities to clean up this repo but that was not the purpose of this PR.

## Misc

> I think as an engineer / contributor to this project I'd like there to be a way to switch between these modes, but it's not clear this would be all that valuable to a reader of the root level docs page. 

I'm not sure what you mean by this, so maybe it's worth discussing in conversation

1egoman replied:
Talked about this on the call, @thomasyuill-livekit is going to add another page that allows a developer to test the script embed version of the popup.

thomasyuill-livekit commented on components/embed-popup/agent-client.tsx:
```diff
@@ -83,115 +88,77 @@ function EmbedFixedAgentClient({
+      <RoomAudioRenderer />
+      <StartAudio label="Start Audio" />
+
+      <Trigger error={!!currentError} popupOpen={popupOpen} onToggle={handleTogglePopup} />
```
I got rid of the backdrop because it prevents the user from interacting with the page while the widget is open.

I imagine a lot of use cases for this widget would benefit from the user still being able to interact with the page while conversing with an agent.

thomasyuill-livekit replied:
If we do want to prevent interaction with the page (which I don't agree we should do), 
we should probably dim the background page by applying a tint to the backdrop .

thomasyuill-livekit commented on components/welcome.tsx:
```diff
@@ -1,88 +1,145 @@
+                transition={{ duration: 0.3, type: 'spring', bounce: 0 }}
+                className="bg-bgAccent/50 dark:bg-bgAccent border-primary/20 dark:border-separatorAccent absolute top-0 h-full w-1/2 rounded-lg border"
+              />
+              <button
+                type="button"
+                onClick={handleClickIframe}
+                className={cn(
+                  'text-fg2 focus:text-fgAccent hover:text-fgAccent z-10 flex-1 cursor-pointer py-2 font-mono transition-colors',
+                  selectedTab === 'iframe' && 'text-fgAccent font-bold'
+                )}
+              >
+                iFrame
+              </button>
+              <button
+                type="button"
+                onClick={handleClickPopover}
+                className={cn(
+                  'text-fg2 focus:text-fgAccent hover:text-fgAccent z-10 flex-1 cursor-pointer py-2 font-mono transition-colors',
+                  selectedTab === 'popover' && 'text-fgAccent font-bold'
+                )}
+              >
+                Popover
+              </button>
```
Yes, I believe this is an artifact of SSR
I'm not sure if we have access to `useSearchParams` until hydration.

I don't believe this is a blocker, but I'll look into it.

thomasyuill-livekit commented on components/embed-iframe/agent-client.tsx:
```diff
@@ -14,6 +15,24 @@ import { WelcomeView } from './welcome-view';
 const MotionWelcomeView = motion.create(WelcomeView);
 const MotionSessionView = motion.create(SessionView);
 
+const THEME_SCRIPT = `
+  const doc = document.documentElement;
+  const theme = new URLSearchParams(window.location.search).get('theme') ?? 'dark';
+
+  if (theme === "system") {
+    if (window.matchMedia("${THEME_MEDIA_QUERY}").matches) {
+      doc.classList.add("dark");
+    } else {
+      doc.classList.add("light");
+    }
+  } else {
+    doc.classList.add(theme);
+  }
+`
+  .trim()
+  .replace(/\n/g, '')
+  .replace(/\s+/g, ' ');
+
```
[next-theme](https://www.npmjs.com/package/next-themes) (used by ShadCN applies similar tomfoolery)
https://github.com/pacocoursey/next-themes/blob/main/next-themes/src/index.tsx#L211-L217

I just feel that library is like using a bazooka to hammer a nail.

1egoman replied:
Ah interesting, cool that makes sense. That being said though, I think as I suggested in the original message, it would be good to use the `ApplyThemeScript` component [here](https://github.com/livekit-examples/agent-starter-embed/pull/8/files#diff-bb1849f01fac76a4d51f01ea8c909881f2f44225ab5df00918797703cbea907bL48) rather than duplicating the script tag outright.

Given this is a project that is meant for users to be able to read and easily understand what is going on, a script tag like that front and center IMO adds a lot of unnecessary confusion (it's a kinda weird pattern) that detracts from ease of understanding.

thomasyuill-livekit replied:
nm, I see what you are getting at now
I'll take a look at revising this approach for the `/embed` route

thomasyuill-livekit replied:
see https://github.com/livekit-examples/agent-starter-embed/pull/8/commits/0e8ff1564bada16423d03baf04f052f89489d782

1egoman commented:
One other small visual thing I noticed - when in dark mode, the overscroll looks kinda weird / makes some of the white text illegible:


https://github.com/user-attachments/assets/e2051957-6060-40bc-bcc7-759f503e8952


Maybe it would be a good idea to wrap the whole welcome page in a fixed div / inset 0 wrapper to stop that from happening?

thomasyuill-livekit commented:
> when in dark mode, the overscroll looks kinda weird
good catch, we need to apply `bg-background` to the body element

thomasyuill-livekit commented:
> I think the same effect could be achieved with a position: fixed; inset: 0; overflow-y: auto; div wrapping the whole welcome page?

I know we do that in the react starter, but that is to solve a different issue. 
It feels a bit icky to me for this use case.

I've refactored in https://github.com/livekit-examples/agent-starter-embed/pull/8/commits/b01dcae1357874540a32d7f3f28a30d72a4775e5 so we can have a reusable `root-layout` component that accepts a `className` and 2 root page groups with separate implementation of `RootLayout`

1egoman commented:
Potential visual issue:

<img width="207" height="159" alt="Screenshot 2025-08-15 at 9 27 49 AM" src="https://github.com/user-attachments/assets/c35ccd7b-f816-494d-b5cd-4e7e8198a390" />

1egoman commented:
Sweet. If you didn't see I approved it but I'll continue to post little visual if I notice them but if you don't get to them I'll make some tickets.

1egoman commented:
One visual thing, which I think I noticed this because you changed `?tab=popover` to `?tab=popup` - there isn't an empty state / it renders a bit weirdly if that query param is set to a non expected value:

<img width="1091" height="575" alt="Screenshot 2025-08-15 at 10 13 04 AM" src="https://github.com/user-attachments/assets/227bd858-80cf-4032-81dc-8cd4fc3b2b32" />

thomasyuill-livekit commented:
> it renders a bit weirdly if that query param is set to a non expected value

resolved, default to `iframe` unless a valid param is passed
